src/repo/mem.rs

```diff
@@ -164,6 +178,29 @@ mod tests {
         assert_eq!(get.await.unwrap(), None);
     }
 
+    #[async_std::test]
+    async fn test_mem_blockstore_list() {
+        let tmp = temp_dir();
+        let mem_store = MemBlockStore::new(tmp.into());
+
+        mem_store.init().await.unwrap();
+        mem_store.open().await.unwrap();
+
+        for data in &[b"1", b"2", b"3"] {
+            let data_slice = data.to_vec().into_boxed_slice();
+            let cid = Cid::new_v1(Codec::Raw, Sha2_256::digest(&data_slice));
+            let block = Block::new(data_slice, cid);
+            mem_store.put(block.clone()).await.unwrap();
+            assert!(mem_store.contains(block.cid()).await.unwrap());
+        }
+
+        let cids = mem_store.list().await.unwrap();
+        assert_eq!(cids.len(), 3);
+        for cid in cids.iter() {
+            assert!(mem_store.contains(cid.as_ref().unwrap()).await.unwrap());
+        }
```

Comment:
You could replace the `cids.iter()` with `cids` (take ownership of the values, as they are not needed after this loop) so that the next line `cid.as_ref().unwrap()` would become `&cid.unwrap()` and after the proposed removal of `Ok(cid...)` this would become `&cid`.